### ISSUE TITLE
exclude_me flag not working as per default

### ISSUE THREAD

**Issue by firasalkafri** [Matthias247/jawampa#77]
Publisher still gets the message without setting the default exclude_me flag to false

**Comment by Matthias247** [comment]
This needs a bit more background info. Are you using jawampa client or server? And with which peer are you testing?

From a short look at the code I don't see an issue:
The client sets exclude_me only to false when the DontExcludeMe flag is set: https://github.com/Matthias247/jawampa/blob/master/jawampa-core/src/main/java/ws/wamp/jawampa/client/SessionEstablishedState.java#L425-L427
The router will by default skip the publisher. Only when the  exclude_me flag is set to false it won't skip it: https://github.com/Matthias247/jawampa/blob/master/jawampa-core/src/main/java/ws/wamp/jawampa/WampRouter.java#L871-L880

**Comment by firasalkafri** [comment]
@Matthias247 I'm using:

    jawampa-core:0.4.0
    netty:0.4.0

With the latest Crossbar server. The client is subscribed to a specific event and when it publishes a message to it the sender still receives the message as if the dontExcludeMe flag was set to true.

**Comment by Matthias247** [comment]
If this is the case then it is a Crossbar issue.
If you don't set DontExcludeMe then jawampa will send the publish message without an `exclude_me: false` option to the router. And the router should then not send the event to the publisher.
Or do you probably have multiple clients? One which calls publish and the other one receives the event? In such a scenario exclude_me would not affect the second client.

**Comment by firasalkafri** [comment]
@Matthias247 Yes I do have multiple clients that are subscribed to the same topic and all of them are supposed to receive the publication right.

I'm just talking about the same client which publishes the message. I will look even more deeper into it and update this thread as soon as possible. For now I suggest keeping it open in case someone has the same issue.
